### ISSUE TITLE
Error flashing "cc3000" or "tinker"

### ISSUE THREAD

**Issue by tslater** [spark/particle-cli#73]
Whenever I try and flash cc3000 or tinker, I get an error that says I'm trying to flash an object instead of a string filepath:

````
Found DFU device 2b04:d006
checking file  [object Object]
running dfu-util -d 2b04:d006 -a 0 -i 0 -s 0x080A0000:leave -D [object Object]
Error writing firmware... Error: Command failed: /bin/sh -c dfu-util -d 2b04:d006 -a 0 -i 0 -s 0x080A0000:leave -D [object Object]
dfu-util: Could not open file [object for reading: No such file or directory
````

**Comment by tslater** [comment]
It looks like they're pulling in the whole object from here (line 67 from settings.js):
````
	knownApps: {
		"deep_update_2014_06": "binaries/deep_update_2014_06.bin",
		"cc3000": {
			'core': "binaries/cc3000-patch-programmer.bin"
		},
		"cc3000_1_14": {
			'core': "binaries/cc3000-patch-programmer_1_14.bin"
		},
		"tinker": 
		{
			'core': "binaries/core_tinker.bin",
			'photon': "binaries/photon_tinker.bin"
		},
		"voodoo": "binaries/voodoospark.bin"
	},
````


Is "cc3000" not available for photon yet?

**Comment by kennethlimcp** [comment]
@tslater ,

the Photon uses a Broadcom Wifi module and the CC3000 is used in the Core. We made this "smart" identification so that users don't perform what you are exactly doing.. Which is to flash the binary not meant for that particular device.

Let us know what issues you are facing and we can help you further.

**Comment by tslater** [comment]
I should be able to flash the tinker but I can't. The cli is trying to use the whole object as the string path.

**Comment by kennethlimcp** [comment]
@tslater , ok the new version got out few hours ago. I will check and send in a PR when i test it. 

Thanks for reporting!

**Comment by tslater** [comment]
of course!

**Comment by kennethlimcp** [comment]
Meanwhile, you can do this:

download: https://github.com/spark/particle-cli/blob/master/binaries/photon_tinker.bin

`particle flash --usb photon_tinker.bin`

If it doesn't work for this version, rollback using `npm install -g particle-cli@1.5.16`

**Comment by emilyrose** [comment]
Hey @tslater, it looks like I published an incomplete update to the way DFU functions. Sorry about that!

I'll post an update and close this issue when I fix it. Thanks!

**Comment by emilyrose** [comment]
@tslater what command are you using exactly when you flash?

**Comment by tslater** [comment]
```particle flash --factory tinker```

**Comment by tslater** [comment]
```particle flash --factory tinker```

**Comment by emilyrose** [comment]
@tslater if you use `particle flash <device name> tinker` it will prompt you for the device type (select Photon), and use the Cloud to flash tinker remotely :smile_cat:

**Comment by tslater** [comment]
ok thanks!

**Comment by emilyrose** [comment]
@tslater absolutely! and thank you for the bug report! I'm working on fixing this issue now :smile:

**Comment by emilyrose** [comment]
ok, this has been fixed in v1.5.19 (59f2edcba6f75d70c080d4fa13833343edbd8485)

thanks again @tslater :smile_cat: I really appreciate it!
